### DISCUSSION TITLE
Dashboard layout designer using grid splitters

### DISCUSSION THREAD

**Discussion by metacoresystems** [CommunityToolkit/Labs-Windows#486] [Experiments]
This initial prototype dashboard layout control makes use of the Community Toolkit grid splitter controls to enable a user to design a responsive resizable layout of panels (a 'dashboard').

The user can split panels in the layout horizontally or vertically, can add a new pane on the top/bottom/left/right, remove a panel and swap panels using a drag manipulation.  You can also serialize or deserialize the layout to DTOs, paving the way for persisting the layout.

An example preview video:

https://github.com/CommunityToolkit/Labs-Windows/assets/114292661/98addf34-cf85-4633-8684-f0163276b11c

The content template for each panel in the layout can be configured by means of a data template selector which selects a template based upon the data context associated with each panel in the layout. In the demonstration app, this is simply a panel that displays a random number.

The code repository containing both a Win UI 3 and Uno/WASM app demonstrating the prototype control is at https://github.com/metacoresystems/CommunityToolkit/tree/main/CommunityToolkit.SplitterPanelLayoutControl

Some additional features that might be added in future:

- Per panel settings/configuration that appears when the control is in design mode
- Serializer implementations for the layout DTOs for out of the box persistence of the layouts
- Make the context menu that appears in the layout customizable

This control particularly fills a functionality gap in the web environment, where historically achieving this sort of functionality in a CSS/HTML environment has been more difficult.

**Comment by moolicc** [comment]
I'm looking for something similar to avalondock for uno, this seems like it'd be a great fit for me personally. Thanks for the work you've put into it so far!
